### ISSUE TITLE
Using existing AI cores and nanites?

### ISSUE THREAD

**Issue by Hyperseeker** [Lorith/AdvancedGear-CDDA-Mod#5]
The game already has AI cores for robots: [a regular one](http://cdda-trunk.chezzo.com/ai_module), [a basic one](http://cdda-trunk.chezzo.com/ai_module_basic), and [an advanced one](http://cdda-trunk.chezzo.com/ai_module_advanced).

There may be little use for the basic one (except maybe for the multitool?), but the regular and advanced ones could serve as basis for the nanotech. Instead of having to wipe/rewrite them prior to creating the required cores, those could be used as is, with the wipe/rewrite being handwaved as already a part of the process of crafting using them. (Introduce the required items into the recipes if necessary.)

There's also [existing nanomaterial](http://cdda-trunk.chezzo.com/nanomaterial) in the game. While it feels raw compared to the mod's current nanites (which feel superengineered using cool sci-fi tech), it may still be converted to the mod's version. Maybe the in-game nanites could be extracted into whatever form the mod's nanites take using some sort of a tool, or the nanofabricator itself could be enhanced to further rearrange the nanostructure into the mod's nanites.

Basically, I'm wondering if you can use the things already in the game, instead of creating replicas for the mod's purposes.

**Comment by Lorith** [comment]
Sorry for taking a while to respond, I don't look at the issues and the like unless someone mentions it on discord, and don't get notifications on github, should probably look into fixing that.

Anyways, the basegame AI cores are both new, and very low power if you consider what they would need to do -- the nanites in this mod are, even in the small quantity, representing countless nanites.  Billions is still a tiny number.  An AI core able to control that many objects would need far more power than something controlling large-scale robotics, even with decent intelligence.  I may change a few of the lower end things to use the advanced core in the future, but not sure.

Likewise, the basegame nanotech stuff came quite a while after I made this mod, and is much less advanced stuff (not in terms of code, but items ingame) and personally I have never encountered it.  I am not against the idea of having the nanomaterial be craftable, as it is basically just another form of feedstock but combined together.

At any rate, it will have to wait a while, as I am currently trying to work on another mod and just not doing very well at it due to lack of motivation.  I'll keep this here for future reference.

**Comment by Hyperseeker** [comment]
For what it's worth, you might consider combining multiple native AI cores into a new one (call it "nanotech AI controller", perhaps) via crafting. It lessens the burden of the mod, in that it adds fewer things that can break the native game mechanics, which is always good to consider. It's an excellent mod that I feel could use a bit of optimization that way.

Either way: best wishes with the new mod, and I hope you don't forget about this one! If the new mod is as bitchin' and cool as this one (and doesn't involve dinosaurs), I'd try it.

**Comment by Lorith** [comment]
Part of the issue is that I have been trying to make sure everything balances out right, which means all the weights match, and as long as I have a single set of components, that works out.  Combining existing AI cores into new ones would lead to huge mass differences, requiring another copy of everything with higher mass.  Additionally, I need my own custom AI cores for the experimental version anyways, so it works out a lot easier to just use the same ones for crafting items.

As to the new mod, not sure if you would consider it good or not, it is a combo of a bunch of monsters, some mutation trees, and magic stuffs.  Unfortunately, CDDA doesn't support editing mutations currently, so it is incompatible with anything else that edits mutations.  Plus I'm horrible at getting it done, making almost no progress lately.  Hint to content: "Touch fluffy tail"

**Comment by Hyperseeker** [comment]
In my experience, if you consistently fail at doing something, it's more likely because you don't like what you're doing. Maybe you like the idea but don't *really* want to work on making it – which is fine. Maybe you want to do something instead.
